src/chunkserver/op_request.cpp

```diff
@@ -749,21 +781,30 @@ void PasteChunkInternalRequest::OnApply(uint64_t index,
 void PasteChunkInternalRequest::OnApplyFromLog(std::shared_ptr<CSDataStore> datastore,  //NOLINT
                                                const ChunkRequest &request,
                                                const butil::IOBuf &data) {
-    // NOTE: 处理过程中优先使用参数传入的datastore/request
-    auto ret = datastore->PasteChunk(request.chunkid(),
-                                     data.to_string().c_str(),
-                                     request.offset(),
-                                     request.size());
-    if (CSErrorCode::Success == ret)
-        return;
+    while (true) {
+        // NOTE: 处理过程中优先使用参数传入的datastore/request
+        auto ret = datastore->PasteChunk(request.chunkid(),
+                                         data.to_string().c_str(),
+                                         request.offset(),
+                                         request.size());
+        if (CSErrorCode::Success == ret)
+            return;
 
-    if (CSErrorCode::InternalError == ret) {
-        LOG(FATAL) << "paste chunk failed: "
-                   << ", request: " << request.ShortDebugString();
-    } else {
-        LOG(ERROR) << "paste chunk failed: "
-                   << ", request: " << request.ShortDebugString();
-    }
+        if (CSErrorCode::InternalError == ret) {
+            LOG(FATAL) << "paste chunk failed: "
+                       << ", request: " << request.ShortDebugString();
+        } else if (CSErrorCode::NoSpaceError == ret) {
+            LOG(ERROR) << "paste chunk failed: "
+                   << ", request: " << request_->ShortDebugString();
+            sleep(WAIT_FOR_DISK_FREED);             
```

Comment:
ditto, use `bthread_usleep` and it's better to add `WAIT_FOR_DISK_FREED` into configuration file like `chunkfilepool.diskUsagePercentLimit`